src/withHook.tsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+type passedHook<T> = (props: React.Props<T>) => object | undefined;
```

Comment:
with such implementation you can't pass hook like `const useNumber = ({ number }: { number: number }) => number;` because `number` not in `object` set